src/analysis/souffle/may_will.dl

```diff
@@ -37,8 +37,23 @@ disallowedUsage(dataConsumer, usage, owner, tag) :-
 .decl allUsagesAllowed(dummy_var: symbol)
 
 .decl isUsage(usage: Usage)
+// DO NOT SUBMIT
+.output isUsage(IO=stdout)
 
-allUsagesAllowed("dummy_var") :- !disallowedUsage(dataConsumer, usage, owner, tag),
-    isPrincipal(dataConsumer), isUsage(usage), isTag(tag), ownsTag(owner, tag).
+isPrincipal(speaker), isPrincipal(actor), isUsage(usage), isTag(tag) :-
+  saysMay(speaker, actor, usage, tag).
+
+isPrincipal(speaker), isUsage(usage), isAccessPath(ap) :- saysWill(speaker, usage, ap).
+
+allUsagesAllowed("dummy_var") :- count : disallowedUsage(_, _, _, _) = 0.
+
+// TODO: Find a better way to do this.
```

Comment:
Not sure if this TODO was meant to be resolved before or after this PR. Opening a comment just to catch the accidental case.

Reply:
It's to be resolved after this PR. But that confusion makes me realize that I should probably open an issue about it.

Reply:
Updated this to point at issue #270 .